src/viam/sdk/common/utils.cpp

```diff
@@ -55,5 +55,22 @@ std::string bytes_to_string(std::vector<unsigned char> const& b) {
     return img_string;
 };
 
+std::chrono::duration<double> duration_from_proto(const google::protobuf::Duration& proto) {
+    return std::chrono::duration<double, std::ratio<1>>(proto.seconds() +
+                                                        proto.nanos() / (double)1e9);
+}
+
+google::protobuf::Duration duration_to_proto(const std::chrono::duration<double>& duration) {
+    google::protobuf::Duration proto;
+    using namespace std::chrono_literals;
+    double total_seconds = duration / 1.0s;
+    long integer_seconds = std::truncl(total_seconds);
+    long integer_nanos = std::truncl(1e9 * (total_seconds - integer_seconds));
+
+    proto.set_seconds(integer_seconds);
+    proto.set_nanos(integer_nanos);
```

Comment:
`std::chrono::duration_from` had rounding errors with my nanosecond tests. This did not have those errors (though this is clearly very hacky so I would be happy to hear suggestions.)